### PR TITLE
17 New Templates Added 

### PR DESCRIPTION
**Pull request by DhiyaneshGeek**

### CODE REVIEW

**Comment by toufik-airane** [comment]
Impressive work! Definitely will look at these templates ASAP.

File: exposures/configs/git-credentials-disclosure.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+id: git-credentials-disclosure
```
Comment (geeknik) [inline exposures/configs/git-credentials-disclosure.yaml:1]: We added a [similar template](https://github.com/geeknik/the-nuclei-templates/blob/main/git-credentials.yaml) to our public repo a couple of months ago, zero false positives thus far.

```
id: git-credentials
info:
  name: Detect .git-credentials
  author: geeknik
  description: Github Authentication
  severity: high
  tags: github,auth

requests:
  - method: GET
    path:
      - "{{BaseURL}}/.git-credentials"

    matchers-condition: and
    matchers:
      - type: word
        part: header
        words:
          - "text/plain"
      - type: word
        words:
          - "https://"
          - "@github.com"
      - type: status
        status:
          - 200
```

**Comment by geeknik** [comment]
Nice. Almost half of those are from our public and private Git repo. 🤣

File: exposures/configs/appsec-yml-disclosure.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - "{{BaseURL}}/appspec.yml"
+      - "{{BaseURL}}/appspec.yaml"
+
+    matchers-condition: and
+    matchers:
+      - type: word
+        words:
+          - "version:"
+          - "os:"
+          - "files:"
```
Comment (toufik-airane) [inline exposures/configs/appsec-yml-disclosure.yaml:22]:
```suggestion
          - "version"
          - "os"
          - "files"
```

Comment (toufik-airane) [reply exposures/configs/appsec-yml-disclosure.yaml:22]: It seems the matcher doesn't include semi-colon :. So should it just be `"\"version\""`, `"\"os\""` and `"\"files\""`?
https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/appsec-yml-disclosure.json#L25

File: exposures/configs/appsec-yml-disclosure.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          - "source:"
```
Comment (toufik-airane) [inline exposures/configs/appsec-yml-disclosure.yaml:23]:
```suggestion
```

Comment (toufik-airane) [reply exposures/configs/appsec-yml-disclosure.yaml:23]: I don't see a `"source"` word in
https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/appsec-yml-disclosure.json.

File: exposures/configs/appsec-yml-disclosure.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        part: body
+        condition: and
+
```
[the remark this answers is not in the document]
Comment (toufik-airane) [reply exposures/configs/appsec-yml-disclosure.yaml:26]: We should check the header.

File: exposures/configs/github-workflows-disclosure.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      - "{{BaseURL}}/.github/workflows/publish.yml"
+      - "{{BaseURL}}/.github/workflows/deploy.yml"
+      - "{{BaseURL}}/.github/workflows/push.yml"
+      - "{{BaseURL}}/.github/workflows/lint.yml"
+      - "{{BaseURL}}/.github/workflows/coverage.yml"
+      - "{{BaseURL}}/.github/workflows/release.yaml"
+      - "{{BaseURL}}/.github/workflows/pr.yml"
+      - "{{BaseURL}}/.github/workflows/automerge.yml"
+      - "{{BaseURL}}/.github/workflows/docker.yml"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/configs/github-workflows-disclosure.yaml:34]:
```suggestion
      - "{{BaseURL}}/.github/workflows/docker.yml"

    matchers-condition: and
```

Comment (toufik-airane) [reply exposures/configs/github-workflows-disclosure.yaml:34]: Add a space.

File: exposures/configs/github-workflows-disclosure.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      - type: regex
+        regex:
+          - "(?m)^\\s*\"?on\"?:"
+          - "(?m)^\\s*\"?jobs\"?:"
+          - "(?m)^\\s*\"?steps\"?:"
+          - "(?m)^\\s*- \"?uses\"?:"
+        part: body
+      - type: word
+        words:
+          - "Content-Type: text/html"
```
Comment (toufik-airane) [inline exposures/configs/github-workflows-disclosure.yaml:45]:
```suggestion
          - "Content-Type: application/yaml"
```

Comment (toufik-airane) [reply exposures/configs/github-workflows-disclosure.yaml:45]: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/github-workflows-disclosure.json#L39

File: exposures/configs/github-workflows-disclosure.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    matchers-condition: and
+    matchers:
+      - type: regex
+        regex:
+          - "(?m)^\\s*\"?on\"?:"
+          - "(?m)^\\s*\"?jobs\"?:"
+          - "(?m)^\\s*\"?steps\"?:"
+          - "(?m)^\\s*- \"?uses\"?:"
+        part: body
+      - type: word
```
Comment (toufik-airane) [inline exposures/configs/github-workflows-disclosure.yaml:43]:
```suggestion
        part: body
        condition: and

      - type: word
```

File: exposures/configs/github-workflows-disclosure.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        words:
+          - "Content-Type: text/html"
+        part: header
+      - type: status
```
Comment (toufik-airane) [inline exposures/configs/github-workflows-disclosure.yaml:47]:
```suggestion
        part: header

      - type: status
```

Comment (toufik-airane) [reply exposures/configs/github-workflows-disclosure.yaml:47]: more space :)

File: exposures/configs/hp-ilo-serial-key-disclosure.yaml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+      - "{{BaseURL}}/xmldata?item=CpqKey"
+    matchers-condition: and
+    matchers:
+      - type: word
+        words:
+          - "LTYPE"
+          - "LNAME"
+          - "KEY"
+        part: body
+      - type: word
```
Comment (toufik-airane) [inline exposures/configs/hp-ilo-serial-key-disclosure.yaml:22]:
```suggestion
        part: body
        condition: and

      - type: word
```

File: exposures/configs/hp-ilo-serial-key-disclosure.yaml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  author: dhiyaneshDk
+  severity: medium
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/hp-ilo-serial-key-disclosure.json
+  tags: exposure,config
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/xmldata?item=CpqKey"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/configs/hp-ilo-serial-key-disclosure.yaml:14]:
```suggestion
      - "{{BaseURL}}/xmldata?item=CpqKey"

    matchers-condition: and
```

File: exposures/configs/hp-ilo-serial-key-disclosure.yaml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        words:
+          - "LTYPE"
+          - "LNAME"
+          - "KEY"
+        part: body
+      - type: word
+        words:
+          - "Content-Type: text/html"
+        part: header
+      - type: status
```
Comment (toufik-airane) [inline exposures/configs/hp-ilo-serial-key-disclosure.yaml:26]:
```suggestion
        part: header

      - type: status
```

File: exposures/configs/kubernetes-kustomization-disclosure.yaml [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  author: dhiyaneshDk
+  severity: medium
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/kubernetes-kustomization-disclosure.json
+  tags: exposure,config
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/kustomization.yml"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/configs/kubernetes-kustomization-disclosure.yaml:14]:
```suggestion
      - "{{BaseURL}}/kustomization.yml"

    matchers-condition: and
```

File: exposures/configs/kubernetes-kustomization-disclosure.yaml [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    matchers:
+      - type: regex
+        regex:
+          - "(?m)^\\s*apiVersion\\s*:\\s*kustomize\\.config"
+          - "(?mi)^\\s*kind\\s*:\\s*Kustomization"
+          - "(?m)^\\scommonLabels\\s*:"
+          - "(?m)^\\sresources\\s*:"
+          - "(?m)^\\snamespace\\s*:"
+        part: body
+      - type: word
```
Comment (toufik-airane) [inline exposures/configs/kubernetes-kustomization-disclosure.yaml:24]:
```suggestion
        part: body

      - type: word
```

File: exposures/configs/kubernetes-kustomization-disclosure.yaml [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+        words:
+          - "Content-Type: application/yaml"
+        part: header
+      - type: status
```
Comment (toufik-airane) [inline exposures/configs/kubernetes-kustomization-disclosure.yaml:28]:
```suggestion
        part: header

      - type: status
```

File: exposures/files/php-user-ini-disclosure.yaml [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+  severity: medium
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/php-user-ini-disclosure.json
+  tags: exposure,files
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/user.ini"
+      - "{{BaseURL}}/.user.ini"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/files/php-user-ini-disclosure.yaml:15]:
```suggestion
      - "{{BaseURL}}/.user.ini"

    matchers-condition: and
```

File: exposures/files/php-user-ini-disclosure.yaml [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+      - type: word
+        words:
+          - "assert"
+          - "highlight"
+          - "opcache"
+          - "mssql"
+          - "oci8"
+          - "agent"
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline exposures/files/php-user-ini-disclosure.yaml:26]:
```suggestion
        part: body
      - type: status
```

Comment (toufik-airane) [inline exposures/files/php-user-ini-disclosure.yaml:26]:
```suggestion
        part: body
        condition: and

      - type: status
```

File: exposures/files/putty-private-key-disclosure.yaml [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+  author: dhiyaneshDk
+  severity: medium
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/putty-private-key-disclosure.json
+  tags: exposure,files
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/my.ppk"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/files/putty-private-key-disclosure.yaml:14]:
```suggestion
      - "{{BaseURL}}/my.ppk"

    matchers-condition: and
```

File: exposures/files/putty-private-key-disclosure.yaml [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+    matchers:
+      - type: word
+        words:
+          - "PuTTY-User-Key-File"
+          - "Encryption:"
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline exposures/files/putty-private-key-disclosure.yaml:21]:
```suggestion
        part: body
        condition: and

      - type: status
```

File: exposures/files/pyproject-disclosure.yaml [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+  author: dhiyaneshDk
+  severity: low
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/pyproject-disclosure.json
+  tags: exposure,files
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/pyproject.toml"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/files/pyproject-disclosure.yaml:14]:
```suggestion
      - "{{BaseURL}}/pyproject.toml"

    matchers-condition: and
```

File: exposures/files/pyproject-disclosure.yaml [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+    matchers:
+      - type: word
+        words:
+          - "[build-system]"
+          - "[tool.poetry]"
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline exposures/files/pyproject-disclosure.yaml:21]:
```suggestion
        part: body
        condition: and

      - type: status
```

File: vulnerabilities/other/nginx-merge-slashes-path-traversal.yaml [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+  name: Nginx Merge Slashes Path Traversal
+  author: dhiyaneshDk
+  severity: medium
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/nginx-merge-slashes-path-traversal.json
+  tags: exposure,config
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}//////../../../etc/passwd"
```
Comment (toufik-airane) [inline vulnerabilities/other/nginx-merge-slashes-path-traversal.yaml:13]:
```suggestion
      - "{{BaseURL}}///////../../../etc/passwd"
```

File: vulnerabilities/other/nginx-merge-slashes-path-traversal.yaml [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+      - "{{BaseURL}}/static//////../../../../etc/passwd"
+      - "{{BaseURL}}///../app.js"
+    matchers-condition: and
```
Comment (toufik-airane) [inline vulnerabilities/other/nginx-merge-slashes-path-traversal.yaml:16]:
```suggestion
      - "{{BaseURL}}///../app.js"

    matchers-condition: and
```

File: vulnerabilities/other/nginx-merge-slashes-path-traversal.yaml [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+    matchers:
+      - type: word
+        words:
+          - "root:"
+          - "app.listen"
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline vulnerabilities/other/nginx-merge-slashes-path-traversal.yaml:23]:
```suggestion
        part: body
        condition: or

      - type: status
```

File: misconfiguration/clockwork-dashboard-exposure.yaml [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+  author: dhiyaneshDk
+  severity: high
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/clockwork-dashboard-exposure.json
+  tags: exposure,unauth
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/__clockwork/latest"
+    matchers-condition: and
```
Comment (toufik-airane) [inline misconfiguration/clockwork-dashboard-exposure.yaml:14]:
```suggestion
      - "{{BaseURL}}/__clockwork/latest"

    matchers-condition: and
```

File: misconfiguration/clockwork-dashboard-exposure.yaml [hunk 25]
```diff
@@ -0,0 +1,28 @@
...
+      - type: word
+        words:
+          - "\"id\""
+          - "\"version\""
+          - "\"method\""
+          - "\"url\""
+          - "\"time\""
+        part: body
+        condition: and
+      - type: word
```
Comment (toufik-airane) [inline misconfiguration/clockwork-dashboard-exposure.yaml:25]:
```suggestion
        condition: and

      - type: word
```

File: exposures/files/ruby-on-rails-secret-token-disclosure.yaml [hunk 26]
```diff
@@ -0,0 +1,23 @@
...
+    path:
+      - "{{BaseURL}}/config/initializers/secret_token.rb"
+    matchers-condition: and
+    matchers:
+      - type: word
+        words:
+          - "secret_key_base ="
+          - "config.secret_token ="
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline exposures/files/ruby-on-rails-secret-token-disclosure.yaml:21]:
```suggestion
        part: body
        condition: and

      - type: status
```

File: exposures/files/snyk-ignore-file-disclosure.yaml [hunk 27]
```diff
@@ -0,0 +1,22 @@
...
+  author: dhiyaneshDk
+  severity: low
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/snyk-ignore-file-disclosure.json
+  tags: exposure,files
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/.snyk"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/files/snyk-ignore-file-disclosure.yaml:14]:
```suggestion
      - "{{BaseURL}}/.snyk"

    matchers-condition: and
```

File: exposures/files/snyk-ignore-file-disclosure.yaml [hunk 28]
```diff
@@ -0,0 +1,22 @@
...
+    matchers:
+      - type: word
+        words:
+          - '# Snyk (https://snyk.io) policy file, patches or ignores known vulnerabilities.'
```
Comment (toufik-airane) [inline exposures/files/snyk-ignore-file-disclosure.yaml:18]:
```suggestion
          - "# Snyk (https://snyk.io) policy file, patches or ignores known vulnerabilities."
```

File: exposures/files/snyk-ignore-file-disclosure.yaml [hunk 29]
```diff
@@ -0,0 +1,22 @@
...
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline exposures/files/snyk-ignore-file-disclosure.yaml:20]:
```suggestion
        part: body

      - type: status
```

File: exposures/files/thumbs-db-disclosure.yaml [hunk 30]
```diff
@@ -0,0 +1,22 @@
...
+  - method: GET
+    path:
+      - "{{BaseURL}}/Thumbs.db"
+    matchers-condition: and
+    matchers:
+      - type: regex
+        regex:
+          - '\\xD0\\xCF\\x11\\xE0\\xA1\\xB1\\x1A\\xE1'
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline exposures/files/thumbs-db-disclosure.yaml:20]:
```suggestion
      - type: binary
        binary:
          - 'D0CF11E0A1B11AE1'
        part: body

      - type: status
```

Comment (toufik-airane) [reply exposures/files/thumbs-db-disclosure.yaml:20]: Have to test it on run-time.

File: exposures/logs/roundcube-log-disclosure.yaml [hunk 31]
```diff
@@ -0,0 +1,23 @@
...
+  severity: low
+  reference: https://github.com/detectify/ugly-duckling/blob/master/modules/crowdsourced/roundcube-log-disclosure.json
+  tags: exposure,logs
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/roundcube/logs/sendmail"
+      - "{{BaseURL}}/roundcube/logs/errors.log"
+    matchers-condition: and
```
Comment (toufik-airane) [inline exposures/logs/roundcube-log-disclosure.yaml:15]:
```suggestion
      - "{{BaseURL}}/roundcube/logs/errors.log"

    matchers-condition: and
```

File: exposures/logs/roundcube-log-disclosure.yaml [hunk 32]
```diff
@@ -0,0 +1,23 @@
...
+    matchers:
+      - type: word
+        words:
+          - "IMAP Error:"
+        part: body
+      - type: status
```
Comment (toufik-airane) [inline exposures/logs/roundcube-log-disclosure.yaml:21]:
```suggestion
        part: body

      - type: status
```

Comment (toufik-airane) [reply exposures/logs/roundcube-log-disclosure.yaml:21]: more spaces 🤤

**Review by toufik-airane — CHANGES_REQUESTED** [review]
Very very very good work! 🔥🔥🔥
I've made some comments and feedback.

Note for the future, by default the `condition` and `matchers-condition` are set to `or`.

Again thank you very much @DhiyaneshGeek for this awesome work. 🚀
And thanks to the https://github.com/detectify/ugly-duckling/ project for the inspiration.

**Comment by DhiyaneshGeek** [comment]
@toufik-airane after accepting the suggestion the YAML validation failed

**Comment by dhayalandeena10** [comment]
help

**Comment by dhayalandeena10** [comment]
made

**Comment by dhayalandeena10** [comment]
new

**Comment by dhayalandeena10** [comment]
2002

-- 
This email and any files transmitted with it are confidential and intended 
solely for the use of the individual or entity to whom they are addressed. 
If you have received this email in error please notify the system manager. 
This message contains confidential information and is intended only for the 
individual named. If you are not the named addressee you should not 
disseminate, distribute or copy this e-mail. Please notify the sender 
immediately by e-mail if you have received this e-mail by mistake and 
delete this e-mail from your system. If you are not the intended recipient 
you are notified that disclosing, copying, distributing or taking any 
action in reliance on the contents of this information is strictly 
prohibited.

**Comment by dhayalandeena10** [comment]
new

**Comment by dhayalandeena10** [comment]
new

File: exposures/configs/dockerfile-hidden-disclosure.yaml [hunk 33]
```diff
@@ -0,0 +1,23 @@
...
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/.dockerfile"
+      - "{{BaseURL}}/.Dockerfile"
+    matchers-condition: and
+    matchers:
+      - type: regex
+        regex:
+          - '^(?:FROM(?:CACHE)?|RUN|ADD|WORKDIR|ENV|EXPOSE|\\#)\\s+[ -~]+'
```
[the remark this answers is not in the document]
Comment (DhiyaneshGeek) [reply exposures/configs/dockerfile-hidden-disclosure.yaml]: @ehsandeep regex updated 
image

File: exposures/configs/git-credentials-disclosure.yaml [hunk 34]
```diff
@@ -0,0 +1,26 @@
...
+  - method: GET
+    path:
+      - "{{BaseURL}}/.git-credentials"
+    matchers-condition: and
+    matchers:
+      - type: regex
+        regex:
+          - "https?://[\\d\\w\\{\\}\\.%]+(?:|@)[\\d\\w\\{\\}\\.]+:[\\d\\w\\{\\}\\.]+(?:|@)[\\d\\w\\{\\}\\.\\?!~#@]+"
+        part: body
+      - type: word
```
[the remark this answers is not in the document]
Comment (ehsandeep) [reply exposures/configs/git-credentials-disclosure.yaml:18]: Yes, it does by adding `negative: true` to any matcher, updated the template accordingly.

```yaml
      - type: word
        part: header
        words:
          - "text/html"
        negative: true
```

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**